src/support/restore-securities.ts

```diff
@@ -1,27 +1,73 @@
-export async function restoreRoles(sdk: any, dump: any) {
+import * as _ from 'lodash'
+
+const sleep = (seconds: number) => new Promise((resolve: any) => setTimeout(resolve, seconds * 1000))
+
+export async function restoreRoles(command: any, dump: any, preserveAnonymous = false) {
   if (dump.type !== 'roles') {
-    throw new Error('Dump file does not contains roles definition')
+    throw new Error('Dump file does not contain roles definition')
+  }
+
+  const anonymousRights = _.get(dump.content, 'anonymous.controllers.*.actions.*')
+
+  if (!preserveAnonymous && anonymousRights === false) {
+    if (command.sdk.username === 'anonymous') {
+      command.logKo('You are currently loggued as "anonymous" and anonymous role write will be overwritten.')
+      command.logInfo('Use the --preserve-anonymous flag to keep default anonymous rights.')
+
+      throw new Error('Please authenticate before importing or use --preserve-anonymous.')
```

Comment:
I don't get why this is a problem with anonymous, and not with any other rights. And I'm not sure that "restoring all roles apart from anonymous" is that useful, as anonymous is usually the first one to be (severely) restricted.

Wouldn't it be better to use `admin:loadSecurities` to prevent that kind of problem, as this route acts as an atomic way to restore roles, profiles and users?

Reply:
IMHO most of the time people use the export/import commands of the CLI to bring the production data into their local setup in order to reproduce a bug.

In production the anonymous user is restricted but in local it's common to keep the default rights.

When you are importing users, roles, profiles and collections with a single `import` command, after loading the anonymous rights every subsequent import will fail because the anonymous user is now restricted.

I feel this is very related to Kourou and the `import` command so that's why I put the logic here and not into Kuzzle.

Reply:
> Also, if you import the roles but you didn't create any user on your local instance then you are doomed.

Oh. Of course. There is that. :expressionless: